istioctl/cmd/istioctl/convert_ingress.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+	"k8s.io/api/extensions/v1beta1"
+
+	"istio.io/istio/istioctl/pkg/convert"
+	"istio.io/istio/pilot/pkg/config/kube/crd"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+)
+
+var (
+	inFilenames        []string
+	outConvertFilename string
+	convertIngressCmd  = &cobra.Command{
+		Use:   "convert-networking-config",
+		Short: "Convert networking configs from v1alpha1 to v1alpha3",
+		Long: "Converts sets of v1alpha1 configs to v1alpha3 equivalents on a best effort basis. " +
```

Comment:
Can you update the docs, examples, etc. here to make it clear this only handles ingress-to-<gateway/virtualservice> conversion. The previous v1alpha1 support has been removed.

Reply:
+1. Also state taht this is a best effort tool and that the translations need to be inspected manually before applying